### ISSUE TITLE
Add "add reference" code action/diagnostics

### ISSUE THREAD

**Issue by matklad** [rust-analyzer/rust-analyzer#6814]
```rust
let mut grid = 0usize;
let mut next = 92usize;
mem::swap(&mut grid, <|>next);
```

->

```rust
let mut grid = 0usize;
let mut next = 92usize;
mem::swap(&mut grid, &mut next);
```

I am not sure if we should produce a diagnostic here, but, even if we don't produce one, this should be available as a code action.

**Comment by lnicola** [comment]
A-AoC? :smile:

**Comment by matklad** [comment]
yup: https://github.com/matklad/aoc2020

**Comment by flodiebold** [comment]
We might want to unify the "wrap in Ok", "remove semicolon" and this into one type mismatch diagnostic that we for now only show when we have a quick fix?

**Comment by matklad** [comment]
Great idea -- this'll reduce the amount of false-positives a lot.

**Comment by Veykril** [comment]
Implemented in https://github.com/rust-analyzer/rust-analyzer/pull/9810
